mlrose/algorithms.py

```diff
@@ -100,9 +103,6 @@ def hill_climb(problem, max_iters=np.inf, restarts=0, init_state=None,
             best_fitness = problem.get_fitness()
             best_state = problem.get_state()
 
-        if curve:
```

Comment:
I still think this is the best place to update the curve, since this way the curve only includes the fitness of states that the algorithm actually moves to.